imap_processing/quality_flags.py

```diff
@@ -0,0 +1,48 @@
+"""Bitwise flagging."""
+
+from enum import IntFlag
+
+
+# TODO: as needed create a message for each name, value pair
+#  for a more verbose description of what the flag indicates
+class ImapQualityFlag(IntFlag):
+    """Base Quality flags."""
+
+    @property
+    def name(self) -> str:
+        """
+        Override the default name property to handle combined flags.
+
+        Returns
+        -------
+        combined_name : str
+            The combined name of the individual flags.
+        """
+        if self._name_ is not None:
+            return self._name_
+
+        members = [member for member in BaseQualityFlags if member & self == member]
+        return "|".join(str(m.name) for m in members if m != BaseQualityFlags.NONE)
```

Comment:
Should this be `self` instead of the subclass values? I don't think the Ultra one will appear right currently.

Suggest pushing the `self.NONE` up into the initial members check as well, which I also think can possibly be shortened to `if self` indicating if any bit is set.
```suggestion
        members = [member for member in self if (member & self == member) and self]
        return "|".join(str(m.name) for m in members)
```